### PR TITLE
Clean up separation of Type vs. NamedType

### PR DESCRIPTION
**Pull request by spawnia**

### CODE REVIEW

**Review by simPod — CHANGES_REQUESTED** [review]

File: src/Type/Definition/FieldDefinition.php [hunk 1]
```diff
@@ -196,38 +198,24 @@ public function assertValid(Type $parentType): void
...
-            $this->resolveFn === null || is_callable($this->resolveFn),
-            sprintf(
-                '%s.%s field resolver must be a function if provided, but got: %s',
-                $parentType->name,
-                $this->name,
-                Utils::printSafe($this->resolveFn)
-            )
-        );
+        /** @var Type&NamedType $type */
+
```
Comment (simPod) [inline src/Type/Definition/FieldDefinition.php:207]:
```suggestion
```

File: src/Type/Definition/NamedTypeTrait.php [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+use ReflectionClass;
+
+use function array_key_exists;
+use function preg_replace;
+
+/**
+ * @see NamedType
+ */
+trait NamedTypeTrait
```
Comment (simPod) [inline src/Type/Definition/NamedTypeTrait.php:15]: `use NameType<how to name all these contents?>`. It reads well then.

File: src/Type/Definition/FieldDefinition.php (hunk 1 above)
Comment (spawnia) [reply src/Type/Definition/FieldDefinition.php:207]: I actually wanted this right under the previous `if` block, as that is the code that proves this type. PHPCS screwed that up.

File: src/Type/Definition/NamedTypeTrait.php (hunk 2 above)
Comment (spawnia) [reply src/Type/Definition/NamedTypeTrait.php:15]: I definitely want to keep `NamedType` in there, as the trait is directly tied to the interface. This pattern approximates multiple inheritance, which is unfortunately not available in PHP. Perhaps `NamedTypeImplementation` will make it more clear?

Comment (simPod) [reply src/Type/Definition/NamedTypeTrait.php:15]: yup 👌🏾

**Review by simPod — APPROVED** [review]

**Pull request merged**